airflow/sensors/external_task.py

```diff
@@ -164,18 +184,23 @@ def poke(self, context, session=None):
         if self.failed_states:
             count_failed = self.get_count(dttm_filter, session, self.failed_states)
 
-        if count_failed == len(dttm_filter):
+        if count_failed > 0:
```

Comment:
Here I am making the assumption that as long as there is at least one external task failure, then we will want to fail the sensor. Though this changes the original behavior, I think this will be a better behavior?

Reply:
Should we add this comment in code?

Reply:
I don't think it's necessary more than an entry in the `UPDATING.md`. The only situation where you will have multiple counts is when the `execution_date_fn` returns more than one execution date to wait for. However, the original behavior will get you into a weird state when only part of the TIs fail, i.e. one fail and one succeeds, resulting in time out. IMHO, I would consider this as a bug